pkgs/by-name/ne/net-cpp/package.nix

```diff
@@ -0,0 +1,118 @@
+{ stdenv
+, lib
+, fetchFromGitLab
+, fetchpatch
+, gitUpdater
+, testers
+, boost
+, cmake
+, curl
+, doxygen
+, graphviz
+, gtest
+, jsoncpp
+, lomiri
+, pkg-config
+, process-cpp
+, properties-cpp
+, python3
+}:
+
+let
+  pythonEnv = python3.withPackages (ps: with ps; [
+    httpbin
+  ]);
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "net-cpp";
+  version = "3.1.0";
+
+  src = fetchFromGitLab {
+    owner = "ubports";
+    repo = "development/core/lib-cpp/net-cpp";
+    rev = finalAttrs.version;
+    hash = "sha256-qXKuFLmtPjdqTcBIM07xbRe3DnP7AzieCy7Tbjtl0uc=";
+  };
+
+  outputs = [
+    "out"
+    "dev"
+    "doc"
+  ];
+
+  patches = [
+    # Enable disabling of Werror
+    # Remove when version > 3.1.0
+    (fetchpatch {
+      name = "0001-net-cpp-Add-ENABLE_WERROR-option.patch";
+      url = "https://gitlab.com/ubports/development/core/lib-cpp/net-cpp/-/commit/0945180aa6dd38245688d5ebc11951b272e93dc4.patch";
+      hash = "sha256-91YuEgV+Q9INN4BJXYwWgKUNHHtUYz3CG+ROTy24GIE=";
+    })
+
+    # Be more lenient with how quickly HTTP test server must be up, for slower hardware / archs
+    (fetchpatch {
+      url = "https://salsa.debian.org/ubports-team/net-cpp/-/raw/941d9eceaa66a06eabb1eb79554548b47d4a60ab/debian/patches/1007_wait-for-flask.patch";
+      hash = "sha256-nsGkZBuqahsg70PLUxn5EluDjmfZ0/wSnOYimfAI4ag=";
+    })
+  ];
+
+  postPatch = if finalAttrs.doCheck then ''
+    # Use wrapped python. Removing just the /usr/bin doesn't seem to work?
+    substituteInPlace tests/httpbin.h.in \
+      --replace '/usr/bin/python3' '${lib.getExe pythonEnv}'
+  '' else ''
+    sed -i CMakeLists.txt -e '/add_subdirectory(tests)/d'
```

Comment:
```suggestion
    substituteInPlace CMakeLists.txt --replace "add_subdirectory(tests)" ""
```

Reply:
Personal preference is `substituteInPlace` when I actually want to just make a change, and `sed` with `/d` or `/a new thing here` for deletions & additions respectively.

Either way, I'll prolly just upstream an option to make the tests optional and `fetchpatch` it instead, i.e. https://gitlab.com/ubports/development/core/lomiri-download-manager/-/merge_requests/23. Just need to find the time to actually do it.

Reply:
I've submitted the patch & `fetchpatch` it now. Reason for `substituteInPlace` has been noted, I'll try to keep it in mind.